src/java.base/share/classes/sun/util/resources/Bundles.java

```diff
@@ -502,15 +502,13 @@ public boolean equals(Object other) {
             if (this == other) {
                 return true;
             }
-            try {
-                final CacheKey otherEntry = (CacheKey)other;
+            if (other instanceof CacheKey otherEntry) {
                 //quick check to see if they are not equal
                 if (hashCodeCache != otherEntry.hashCodeCache) {
                     return false;
                 }
-                return locale.equals(otherEntry.locale)
-                        && name.equals(otherEntry.name);
-            } catch (NullPointerException | ClassCastException e) {
+                return Objects.equals(locale, otherEntry.locale)
```

Comment:
While the proposal is not wrong, you might want to consider reverting the change to those two lines. The reason is that neither `locale` nor `name` can be null, if I understand this class correctly. 

NPE was seemingly caught to cover for `other` being null.

Reply:
> NPE was seemingly caught to cover for other being null.
Unlike that of other file in this PR.

Reply:
Done